### PR TITLE
[DOCS] Document pc.AnimationComponent#skeleton and pc.AnimationComponent#animations

### PR DESCRIPTION
**Pull request by aidinabedi**
Currently, there is no documented or type-safe way to get the `pc.Skeleton` associated with `pc.AnimationComponent`. This is necessary when exporting PlayCanvas to glTF. This PR fixes that.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
@slimbuck You're the decider.

Just to throw out my opinion, this should really be `pc.AnimationComponent#skeleton` (property instead of function).

**Comment by aidinabedi** [comment]
@willeastcott Fixed.

**Comment by aidinabedi** [comment]
@willeastcott Interestingly, `pc.AnimationComponent#skeleton` already exists it seems. It wasn't super obvious but [apparently it's added by `AnimationComponentSystem`](https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/system.js#L47). So I've changed the PR to just add jsdoc comments, documenting both property `skeleton` and `animations`.

**Comment by aidinabedi** [comment]
Have you had a chance to review this @slimbuck?

File: src/framework/components/animation/component.js [hunk 1]
```diff
@@ -13,7 +13,9 @@ Object.assign(pc, function () {
      * @property {boolean} activate If true the first animation asset will begin playing when the scene is loaded.
      * @property {pc.Asset[]|number[]} assets The array of animation assets - can also be an array of asset ids.
      * @property {number} currentTime Get or Set the current time position (in seconds) of the animation.
-     * @property {number} duration Get the duration in seconds of the current animation.
+     * @property {number} duration Get the duration in seconds of the current animation. [read only]
+     * @property {pc.Skeleton} skeleton Get or Set the skeleton for the current model.
+     * @property {object<string, pc.Animation>} animations Get or Set dictionary of animations by name.
```
Comment (slimbuck) [inline src/framework/components/animation/component.js:18]: The animation component was recently extended to support GLB animations. The result is unfortunately a bit messy: the component is either internally in the legacy 'json' animation mode or the newer 'glb' mode. In the legacy mode skeleton is valid, but in the newer glb mode skeleton is unused.

So it's probably not safe to document skeleton just yet (in any case it's a read-only property).

animations property is also read only.

Comment (aidinabedi) [reply src/framework/components/animation/component.js:18]: @slimbuckThank you for the feedback.

What's the equivalent to `skeleton` in "glb" mode? We're writing a glTF exporter and need to be able to properly access it, in order to export it.

I'm happy to change the documentation to read-only. But please note that both `skeleton` and `animations` have setters - according to [the schema](https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/system.js#L2) which [`_buildAccessors` reads](https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/system.js#L47) to create standard [getters and setters](https://github.com/playcanvas/engine/blob/master/src/framework/components/component.js#L34). Additionally, `AnimationComponent` even [listens for and handles changes to `animations` explicitly](https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/component.js#L24).

Do you want me to still change both to read-only? I'll also add a comment about `skeleton` not being available when the asset is from glb/gltf.

Comment (slimbuck) [reply src/framework/components/animation/component.js:18]: Hi @aidinabedi yup you're right, I forgot about the set_animations handler. It is clearly not read-only!

I guess we can keep skeleton with a comment as you suggest.

GLB mode uses the AnimController and related classes (see anim.js). These are all currently private and some of the API is likely to change in future. So we can't document them just yet.

**Comment by aidinabedi** [comment]
Btw @slimbuck, I have to ask:

How come glTF is often referred to as GLB in PlayCanvas (i.e. `GlbParser` etc)? That confused me when I first read [the PR description that added glTF support](https://github.com/playcanvas/engine/pull/1904#issue-384917384). It gives the impression that only the glTF binary format is supported.

**Comment by slimbuck** [comment]
You're not the first to find this confusing!

GLB is a container format for glTF files and the associated binary data. So glb necessarily includes support for glTF. However the glTF format doesn't necessarily include glb.

This was the mental model I took from the get go, but perhaps it would have been less confusing using the glTF naming instead.

**Comment by aidinabedi** [comment]
@slimbuck You're correct, the container format is often referred to as GLB, [while other times as just `Binary glTF` in the specs](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#glb-file-format-specification):
> To solve this, a container format, Binary glTF is introduced. In Binary glTF, a glTF asset (JSON, .bin, and images) can be stored in a binary blob.

However, both file formats (the container and JSON) are part of the same overall delivery specification, which most people know as glTF.

**Pull request merged**